tools/substitute_tar_deps.bzl

```diff
@@ -0,0 +1,81 @@
+# Copyright Google Inc. All Rights Reserved.
+#
+# Use of this source code is governed by an MIT-style license that can be
+# found in the LICENSE file at https://angular.io/license
+load("@aspect_bazel_lib//lib:jq.bzl", "jq")
+load("@aspect_bazel_lib//lib:utils.bzl", "to_label")
+
+def substitute_tar_deps(name, src, pkg_deps, out):
+    """Substitute tar paths into a package.json file for the packages it depends on.
+
+    Args:
```

Comment:
"src and pkg_deps must be labels in the bazel-out tree for the derived path to the npm_package_archive.tar.gz to be correct"